sw/airborne/firmwares/fixedwing/stabilization/stabilization_adaptive.c

```diff
@@ -520,3 +596,103 @@ inline static void h_ctl_pitch_loop(void)
   h_ctl_elevator_setpoint = TRIM_PPRZ(cmd);
 }
 
+/* yaw damper */
+#if H_CTL_YAW_LOOP
+inline static void h_ctl_yaw_loop(void)
+{
+ 
+#if H_CTL_YAW_TRIM_BY
+  // Actual Acceleration from IMU:
+ #ifndef SITL
+  struct Int32Vect3 accel_meas_body;
+  struct Int32RMat *body_to_imu_rmat = orientationGetRMat_i(&imu.body_to_imu);
+  int32_rmat_transp_vmult(&accel_meas_body, body_to_imu_rmat, &imu.accel);
+  float by = ACCEL_FLOAT_OF_BFP(accel_meas_body.y);
+ #else
+  float by = 0;
+ #endif
+
+  if (pprz_mode == PPRZ_MODE_MANUAL || launch == 0) {
+    h_ctl_yaw_by_sum_err = 0.;
+  } else {
+     if (h_ctl_pitch_igain > 0.){
+//     only update wehn: phi<60degrees and by<2g
+       if(fabs(stateGetNedToBodyEulers_f()->phi) < 1.05 &&
+          fabs(by) < 20.) {
+       h_ctl_yaw_by_sum_err += by * H_CTL_REF_DT;
+//     max half rudder deflection for trim
+       BoundAbs(h_ctl_yaw_by_sum_err, MAX_PPRZ/(2* h_ctl_yaw_by_igain));
+       }
+   } else {
+      h_ctl_pitch_sum_err = 0.;
+    }
+  }
+#endif
+
+float Vo = *stateGetAirspeed_f();
```

Comment:
If you activate the yaw control without the airspeed, it may result in bad value for `Vo`, most likely a saturation to STALL_AIRSPEED.
Maybe a constant "nominal" airspeed should be used if airspeed is not available.